src/ORM/Query.php

```diff
@@ -203,7 +202,7 @@ public function select($fields = [], $overwrite = false)
      *
      * This method returns the same query object for chaining.
      *
-     * @param \Cake\ORM\Table $table The table to pull types from
+     * @param \Cake\Datasource\RepositoryInterface $table The table to pull types from
      * @return $this
      */
     public function addDefaultTypes(Table $table)
```

Comment:
Changing type hints doesn't allow for backwards compatibility as userland subclasses will trigger errors.